### PR TITLE
Add normalization to Lie group

### PR DESCRIPTION
**Pull request by fantaosha**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by luisenp — CHANGES_REQUESTED** [review]
LGTM. We should add some simple tests for non-euclidean groups. For example, start with a bunch of random tensors, normalize, then check that group check holds; do this for batch size 1 and > 1.

File: theseus/geometry/vector.py [hunk 1]
```diff
@@ -213,6 +213,13 @@ def exp_map(
 
         return Vector(data=tangent_vector.clone())
 
+    @staticmethod
+    def normalize(matrix: torch.Tensor) -> torch.Tensor:
```
[the remark this answers is not in the document]
Comment (fantaosha) [reply theseus/geometry/vector.py]: Replaced `matrix` with `data`.

File: theseus/geometry/manifold.py [hunk 2]
```diff
@@ -81,6 +81,14 @@ def project(
     ) -> torch.Tensor:
         return self._project_impl(euclidean_grad, is_sparse)
 
+    @staticmethod
+    def normalize(data: torch.Tensor) -> torch.Tensor:
```
Comment (mhmukadam) [inline theseus/geometry/manifold.py:85]: Is normalize supposed to be `@abc.abstractmethod` with the `Manifold` version doing `pass`?

Comment (mhmukadam) [reply theseus/geometry/manifold.py:85]: Same question for `_data_check`.

**Pull request merged**